src/praisonai-agents/praisonaiagents/context/optimizer.py

```
@@ -576,6 +577,222 @@ def _llm_summarize(self, messages: List[Dict[str, Any]], max_tokens: int) -> str
+                         ["working on", "starting", "will now", "let me", "i'll"]):
+                    tasks_in_progress.append(content[:150] + "..." if len(content) > 150 else content)
+                
+                if any(decision in content_lower for decision in 
+                       ["decided", "chosen", "will use", "approach"]):
+                    key_decisions.append(content[:100] + "..." if len(content) > 100 else content)
+        
+        # Build structured summary
+        summary_parts.append(f"**Conversation Summary** ({len(messages)} messages compressed)")
+        
+        if tasks_completed:
+            summary_parts.append(f"**Completed Tasks:**")
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)